src/librustc_passes/ast_validation.rs

```diff
@@ -167,11 +168,58 @@ impl<'a> AstValidator<'a> {
                 "only lifetime parameters can be used in this context");
         }
     }
+
+    /// With eRFC 2497, we need to check whether an expression is ambigious and warn or error
+    /// depending on the edition, this function handles that.
+    fn while_if_let_ambiguity(&self, expr: &P<Expr>) {
+        if let Some((span, op_kind)) = self.while_if_let_expr_ambiguity(&expr) {
+            let mut err = self.err_handler().struct_span_err(
+                span, &format!("ambigious use of `{}`", op_kind.to_string())
+            );
+
+            err.note(
+                "This will be a error until the `let_chains` feature is stabilized."
+            );
+
+            if let Ok(snippet) = self.session.source_map().span_to_snippet(span) {
+                err.span_suggestion(
+                    span, "consider adding parenthesis", format!("({})", snippet),
```

Comment:
The plural [is](https://en.wiktionary.org/wiki/parentheses) "parenthes**e**s".

Reply:
Thanks, fixed!